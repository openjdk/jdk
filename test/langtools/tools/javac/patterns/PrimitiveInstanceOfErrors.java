/*
 * @test /nodynamiccopyright/
 * @bug 8304487
 * @summary Compiler Implementation for Primitive types in patterns, instanceof, and switch (Preview)
 * @enablePreview
 * @compile/fail/ref=PrimitiveInstanceOfErrors.out -XDrawDiagnostics -XDshould-stop.at=FLOW PrimitiveInstanceOfErrors.java
 */
public class PrimitiveInstanceOfErrors {
    public static boolean unboxingAndNarrowingPrimitiveNotAllowedPerCastingConversion() {
        Long l_within_int_range = 42L;
        Long l_outside_int_range = 999999999999999999L;

        return l_within_int_range instanceof int && !(l_outside_int_range instanceof int);
    }

    public static <T extends Integer> boolean wideningReferenceConversionUnboxingAndNarrowingPrimitive(T i) {
        return i instanceof byte;
    }

    public static void boxingConversionsBetweenIncompatibleTypes() {
        int i = 42;

        boolean ret1 = i instanceof Integer; // (Integer) i // OK and true
        boolean ret2 = i instanceof Double;  // error: incompatible types
        boolean ret3 = i instanceof Short;   // error: incompatible types
    }
}
