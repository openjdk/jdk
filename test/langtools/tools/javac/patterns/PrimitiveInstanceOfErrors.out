PrimitiveInstanceOfErrors.java:13:16: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.Long, int)
PrimitiveInstanceOfErrors.java:13:55: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.Long, int)
PrimitiveInstanceOfErrors.java:17:16: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: T, byte)
PrimitiveInstanceOfErrors.java:24:24: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: int, java.lang.Double)
PrimitiveInstanceOfErrors.java:25:24: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: int, java.lang.Short)
- compiler.note.preview.filename: PrimitiveInstanceOfErrors.java, DEFAULT
- compiler.note.preview.recompile
5 errors